ste/xfer-anyToRemote-file.go

```diff
@@ -363,12 +365,15 @@ func epilogueWithCleanupSendToRemote(jptm IJobPartTransferMgr, s sender, sip ISo
 		_, isS2SCopier := s.(s2sCopier)
 		destLength, err := s.GetDestinationLength()
 
-		if err != nil {
+		if stgErr, stgErrOk := err.(azfile.StorageError); stgErrOk && stgErr.Response() != nil {
```

Comment:
It's probably cleaner, and better idomatic Go, to use pipeline.Response instead of azfile.StorageError.  Why? Because this code doesn't just run for Azure Files, it will also run for Blobs and ADLS Gen 2 (which is known as blobFS in the AzCopy codebase).

You'd probably get away with the existing code, because azfile.StorageError has the same structure as azblob.StorageError and azblobfs.StorageError, but it's still a bit messy. It's cleaner to just use a "narrower" interface, with only the method you need. As it happens, all those error types implement pipeline.Response, which contains exactly the one method that you need.  So using pipeline.Response would be cleaner and (I think) better idiomatic Go.